document/js-api/index.bs

```diff
@@ -1329,6 +1383,134 @@ The algorithm <dfn>ToWebAssemblyValue</dfn>(|v|, |type|) coerces a JavaScript va
 
 </div>
 
+<h3 id="jspi">JavaScript Promise Integration</h3>
+
+Note: The JavaScript Promise Integration API (JSPI) allows WebAssembly functions to suspend and resume their execution -- based on the behavior of JavaScript functions that return Promise objects.
+
+A {{Suspending}} object represents a JavaScript function whose calls via WebAssembly imports should be *suspended* when they return a Promise object.
+Each {{Suspending}} object has a \[[wrappedFunction]] internal slot which holds a JavaScript function.
+
+In addition, the {{promising}} function takes as argument a WebAssembly function and returns a JavaScript function that returns a Promise that is resolved when the WebAssembly function completes.
+
+Each [=agent=] maintains a [=Execution Context Status map=], mapping from [=execution context|execution contexts=] to a status symbol. If present, a status can be one of two <dfn>stack status</dfn> values:
+
+   * <dfn>active</dfn>. This signals that the associated execution context is actively executing and has the potential to be [=paused=].
+   * <dfn>paused</dfn>. This signals that the associated execution is not currently involved in computation and has been paused.
+
+If an execution context is not present in the status mapping, then it may not be [=Pause|paused=] or [=Enter|reentered=].
+
+When a new agent is created, its status mapping is set to the empty map.
+
+<pre class="idl">
+[Exposed=*]
+partial namespace WebAssembly {
+    Function promising(Function wasmFunc);
+};
+
+[LegacyNamespace=WebAssembly, Exposed=*]
+interface Suspending {
+    constructor(Function jsFun);
+};
+</pre>
+
+<div algorithm>
+    The  <dfn method for="WebAssembly">promising(|wasmFunc|)</dfn> function, when invoked, performs the following steps:
+    1. If [$IsCallable$](|wasmFunc|) is false, throw a {{TypeError}}.
+    1. If |wasmFunc| does not have a \[[FunctionAddress]] internal slot, throw a {{TypeError}}.
+    1. Let |builder| be a new [=AbstractClosure=] with no parameters that captures |wasmFunc| and performs [=run a Promising function=] when called.
+    1. Returns the result of [=CreateBuiltinFunction=](|builder|,1,"", « »).
+</div>
+
+<div algorithm>
+  The algorithm to <dfn>run a Promising function</dfn> from the JavaScript object |wasmFunc| and a [=list=] of [=WebAssembly values=] |arguments| consists of the following steps:
+    1. Let |promise| be a new [=PromiseCapabilityRecord=].
+    1. Let |funcaddr| be the value of |wasmFunc|'s \[[FunctionAddress]] internal slot.
+    1. Let |runner| be a new [=AbstractClosure=] with no parameters that captures |promise|, |funcaddr|, and |arguments| and performs [=evaluate a Promising function=](|promise|,|funcaddr|,|arguments|) when called.
```

Comment:
Editorially this should be "and performs the following steps when called:", with the "Perform evaluate a Promising function ..." in a separate stpe, instead of inline.

Reply:
Ack